packages/market/contracts/Trading/MarketLogic.sol

```diff
@@ -1,39 +1,29 @@
-// Copyright 2018 Energy Web Foundation
-// This file is part of the Origin Application brought to you by the Energy Web Foundation,
-// a global non-profit organization focused on accelerating blockchain technology across the energy sector,
-// incorporated in Zug, Switzerland.
-//
-// The Origin Application is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-// This is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY and without an implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details, at <http://www.gnu.org/licenses/>.
-//
-// @authors: slock.it GmbH; Martin Kuechler, martin.kuchler@slock.it; Heiko Burkhardt, heiko.burkhardt@slock.it
-
 pragma solidity ^0.5.2;
 pragma experimental ABIEncoderV2;
 
+import "@energyweb/origin/contracts/Origin/CertificateLogic.sol";
+import "@energyweb/origin/contracts/Origin/TradableEntityContract.sol";
+import "@energyweb/origin/contracts/Interfaces/TradableEntityDBInterface.sol";
+import "@energyweb/origin/contracts/Interfaces/TradableEntityInterface.sol";
+import "@energyweb/asset-registry/contracts/Interfaces/AssetGeneralInterface.sol";
 import "../../contracts/Trading/MarketDB.sol";
 import "../../contracts/Trading/AgreementLogic.sol";
-import "@energyweb/asset-registry/contracts/Interfaces/AssetGeneralInterface.sol";
 
 /// @title The logic contract for the AgreementDB of Origin list
 contract MarketLogic is AgreementLogic {
     event createdNewDemand(address _sender, uint indexed _demandId);
     event createdNewSupply(address _sender, uint indexed _supplyId);
     event DemandStatusChanged(address _sender, uint indexed _demandId, uint16 indexed _status);
     event DemandUpdated(uint indexed _demandId);
+    event DemandFilled(uint indexed _demandId, uint indexed _entityId);
```

Comment:
What is the means of this event? Fully filled? And if yes, what about partial fills? I think we should support demand being filled by multiple certificates.

Reply:
Probably the naming is not clear enough. This is just an event that says that a certificate has been matched to a demand (partially filled).

Reply:
Can we re-construct using this event the amount of energy that was matched during matching event?